packages/devtools_app/lib/src/debugger/flutter/console.dart

```diff
@@ -30,7 +33,102 @@ class Console extends StatefulWidget {
 }
 
 class _ConsoleState extends State<Console> {
-  final scrollController = ScrollController();
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<List<String>>(
+      valueListenable: widget.controller.stdio,
+      builder: (context, lines, _) {
+        // Extract this to a reusable ConsoleWithControls StatelessWidget?
+        return Stack(
+          children: [
+            ConsoleOutput(lines: lines),
+            Container(
```

Comment:
instead of adding the disabled parameter to ConsoleControls, write `if (lines.isNotEmpty)` here, and then the container will only be built if that condition is met

Reply:
Very good point, will do!